pkg/router/routeapihelpers/validation.go

```diff
@@ -349,6 +349,11 @@ func validateCertificatePEM(certPEM string, options *x509.VerifyOptions) ([]*x50
 		return nil, fmt.Errorf("invalid/empty certificate data")
 	}
 
+	// Reject any MD5-based certs as HaProxy will refuse to start with them.
+	if certs[0].SignatureAlgorithm == x509.MD5WithRSA {
```

Comment:
Now i wonder, since we know there is a problem with SHA1 and MD5, maybe we should just come up with a list of what HAProxy **accepts**, instead of adding one by one each old algorithm that will break HAProxy?

Reply:
Hmm let me think about that.

- An accept list (opt-in) vs. a reject list (opt-out) produce the same result.
- If there are new algorithms added, an accept list would add a bit of maintenance burden since we'd need to go and add new algorithms, which could potentially prevent customers from using a new algorithm that HaProxy supports.
- An accept list doesn't change the fact that we may need to deprecate algorithms in the future, like we are doing now (we'd need to remove it from the accept list instead of adding to reject list).
- The accept list (~13) is larger than the reject list (~3)

I can handle it in either manner, but I'm struggling to see the benefit to switching to an accept list. It seems essentially the same, but arguably adds more maintenance. Am I misunderstanding a benefit of doing an accept list?

Reply:
I'm coming from the viewpoint that over the next few years we'll need to remove more algorithms than we'll need to add.  I'm okay with a wait and see approach on this.

Reply:
One last question - if we have 3 signature algorithms in the reject list, why not add the other two here?

Reply:
Yea I should be more clear: the other two (SHA1 RSA and SHA1 ECDSA) need to come later via https://github.com/openshift/router/pull/552 because they are supported in 4.15, but not in 4.16. I'm keeping them separated, because this PR needs backporting, while SHA1 can't be backported.

I updated to be switch statement for easier expansion later.